pandas/core/internals/blocks.py

```diff
@@ -2050,19 +2050,23 @@ def to_native_types(
 ) -> np.ndarray:
     """ convert to our native types format """
     values = ensure_wrapped_if_datetimelike(values)
+    if isinstance(values, Categorical) and isinstance(values.categories[0], Timestamp):
```

Comment:
You should be able to use `.dtype` to get the category type in a more stable way. Also, why is this timestamp specific? Do other datetimelike values work (and are tested?)